File: cpp/gms/gms-filters-provider/src/parameters/FIRFilterParameters.cpp
#include "FIRFilterParameters.hh"

FIRFilterParameters::FIRFilterParameters(int groupDelaySec,
                                         bool isDesigned,
                                         double sampleRateHz,
                                         double sampleRateToleranceHz,
                                         std::vector<double> transferFunctionB) : transferFunctionB(transferFunctionB),
                                                                                  BaseFilterParameters(groupDelaySec,
                                                                                                       isDesigned, 
                                                                                                       sampleRateHz, 
                                                                                                       sampleRateToleranceHz)
{
  this->numTransferFunction = transferFunctionB.size() / 3;
};
#if (__EMSCRIPTEN__)

emscripten::val FIRFilterParameters::getTransferFunctionAsTypedArray()
{
  return emscripten::val(emscripten::typed_memory_view(transferFunctionB.size(), transferFunctionB.data()));
}

#endif

File: typescript/user-interface/packages/ui-wasm/src/ts/gms-filters-provider/gms-filters/types/fir-filter-parameters.ts
import type { WasmVectorDouble } from './vector-double';

export interface WasmFIRFilterParametersModule {
  new (
    groupDelaySec: number,
    isDesigned: boolean,
    sampleRateHz: number,
    sampleRateToleranceHz: number,
    transferFunctionB: WasmVectorDouble
  ): WasmFIRFilterParameters;
}
export interface WasmFIRFilterParameters {
  isDesigned: boolean;
  groupDelay: number;
  $$: {
    ptr: number;
  };
  delete: () => void;
  getTransferFunctionAsTypedArray: () => number[];
}
